### PR TITLE
PR for adding notification and friends system

### PR DESCRIPTION
**Pull request by iAmao**
#### What does this PR do?

This PR add a notification system and friends system

#### Description of Task to be completed?

This PR allows in house notification system. I.e. user gets a notification that is equivalent to clicking the invite link when an invite is sent. User can now search for people within the game and send them a friend request, which the recipient can either accept or reject the request.

#### How should this be manually tested?

- clone the repo : `git clone https://github.com/andela/izanami-cfh.git`
- install dependencies: `npm install`
- start the app: `npm start`

#### Any background context you want to provide?

N/A

#### What are the relevant pivotal tracker stories?

#139456273 Registered users should be able to invite other users to connect as "friends"

#### Screenshots (if appropriate)

N/A

#### Questions:

N/A

### CODE REVIEW

**Review by eshaibu — APPROVED** [review]

File: config/socket/socket.js [hunk 1]
```diff
@@ -102,6 +168,7 @@ module.exports = function (io) {
           player.premium = user.premium || 0;
           player.avatar = user.avatar || avatars[Math.floor(Math.random() * 4) + 12];
         }
+        console.log('[GAME ROOM IS]', data.room);
```
Comment (ooduye) [inline config/socket/socket.js]: Why console log

File: config/socket/socket.js [hunk 2]
```diff
@@ -140,6 +207,7 @@ module.exports = function (io) {
           gamesNeedingPlayers.shift();
           game.prepareGame();
         }
+        // socket.emit('sendCustomInvite', { gameID: game.gameID });
```
Comment (ooduye) [inline config/socket/socket.js]: Commented out code...

File: public/js/controllers/game.js [hunk 3]
```diff
@@ -191,49 +192,60 @@ angular.module('mean.system')
...
           $scope.inviteeUserEmail = '';
           $scope.inviteeUserName = '';
         }
       };
 
-      $scope.playerSearch = () => {
-        if ($scope.inviteeUserName !== '') {
-          playerSearch.getPlayers($scope.inviteeUserName).then((data) => {
-            $scope.searchUserResults = data;
+    $scope.playerSearch = () => {
```
Comment (ooduye) [inline public/js/controllers/game.js]: Indentation issue

**Comment by iAmao** [comment]
@andela-ooduye violations are fixed

**Pull request merged**